### ISSUE TITLE
sync:files fails due to non-interactive shell

### ISSUE THREAD

**Issue by danepowell** [acquia/blt#1874]
My system information:
* Operating system type: Ubuntu
* Operating system version: 17.04
* BLT version: 8.9

Output of `blt doctor`:
all green

When I run this command:
```
blt sync:files
```

I get the following output:
```
[Acquia\Blt\Robo\Tasks\DrushTask] Running /var/www/foo/vendor/bin/drush rsync '@foo.test:%files' /var/www/foo/docroot/sites/default/files --exclude-paths='styles:css:js' in /var/www/foo/docroot
You will delete files in /var/www/foo/docroot/sites/default/files and replace with data from foo.test@foo:/var/www/html/foo.test/docroot/sites/default/files
Do you really want to continue? (y/n): Cancelled.
[Acquia\Blt\Robo\Tasks\DrushTask]  Exit code 75  Time 2.775s
```

And I expected this to happen:
Files should be synced

Like the litany of recent issues related to sync:refresh more generally, this seems to be due to Robo running the commands in a non-interactive shell.

**Comment by grasmash** [comment]
Can you please share the full output, using –vvv,  and also the full output of `blt doctor`?

Does this only happen in the VM?  I am wondering if we could have something to do with your version of vagrant or the vagrant exec plug-in.

I've crossed out the possibility that this has to do with PHP versions or with your shell. However, this does have something to do with your specific machine. This issue only affects some people and not others, even when using the same exact code base.

Would you be able to step through the execution via XDebug?

**Comment by danepowell** [comment]
I only use VMs. It makes sense that this would be a problem, I bet vagrant exec runs non-interactively and won't forward shell prompts.

**Comment by grasmash** [comment]
After some investigation, I determined that this is an issue with vagrant or the vagrant exec plugin. My previous version, 1.9.1, worked fine. The newest version, 1.9.7. causes this issue.

It seems that when vagrants exec is used to execute a comma inside of the virtual machine, the terminal is non-interactive. This prevents drush (any other library) to provide the user with a prompt. 

I have introduced #1880 as a solution. This will automatically assume –y for all drush commands on non-interactive shells.
